flocker/provision/_ssh/_conch.py

```diff
@@ -131,8 +131,12 @@ def perform_run(dispatcher, intent):
             message_type="flocker.provision.ssh:run",
             command=intent.log_command_filter(intent.command),
         ).write()
+        if hasattr(intent, 'pre_command') and intent.pre_command:
```

Comment:
@jml The only situation where `hasattr` can act weirdly is if the object has a weird (probably stateful) `__getattr__`/`__getattribute__` handler.  Nevertheless, I was dithering on these two versions myself, so happy to change it.